--- rustc-1.44.1-src/src/tools/cargo/Cargo.toml	2020-06-18 05:17:31.000000000 +0900
+++ patches/rustc-1.44.1-src/src/tools/cargo/Cargo.toml	2020-07-02 23:51:29.000000000 +0900
@@ -75,7 +75,7 @@
 # for more information.
 rustc-workspace-hack = "1.0.0"
 
-[target.'cfg(target_os = "macos")'.dependencies]
+[target.'cfg(any(target_os = "macos", target_os = "ios"))'.dependencies]
 core-foundation = { version = "0.7.0", features = ["mac_os_10_7_support"] }
 
 [target.'cfg(windows)'.dependencies]
