--- a/src/loslib.c
+++ b/src/loslib.c
@@ -132,8 +132,9 @@
 
 #if !defined(l_system)
 #if defined(LUA_USE_IOS)
-/* Despite claiming to be ISO C, iOS does not implement 'system'. */
-#define l_system(cmd) ((cmd) == NULL ? 0 : -1)
+/* iOS SDK hides system(3). Mayflower links mayflower_system (libiosexec). */
+extern int mayflower_system(const char *);
+#define l_system(cmd) ((cmd) == NULL ? 1 : mayflower_system(cmd))
 #else
 #define l_system(cmd)	system(cmd)  /* default definition */
 #endif
