--- rustc-1.44.1-src/src/tools/build-manifest/src/main.rs	2020-06-18 05:17:06.000000000 +0900
+++ patches/rustc-1.44.1-src/src/tools/build-manifest/src/main.rs	2020-07-01 10:00:22.000000000 +0900
@@ -18,10 +18,13 @@
 use std::process::{Command, Stdio};
 
 static HOSTS: &[&str] = &[
+    "aarch64-apple-ios",
     "aarch64-unknown-linux-gnu",
     "arm-unknown-linux-gnueabi",
     "arm-unknown-linux-gnueabihf",
+    "armv7-apple-ios",
     "armv7-unknown-linux-gnueabihf",
+    "armv7s-apple-ios",
     "i686-apple-darwin",
     "i686-pc-windows-gnu",
     "i686-pc-windows-msvc",
@@ -149,6 +152,9 @@
 ];
 
 static DOCS_TARGETS: &[&str] = &[
+    "aarch64-apple-ios",
+    "armv7-apple-ios",
+    "armv7s-apple-ios",
     "i686-apple-darwin",
     "i686-pc-windows-gnu",
     "i686-pc-windows-msvc",
