--- rustc-1.44.1-src/src/llvm-project/llvm/include/llvm/Support/Compiler.h	2020-07-05 21:47:40.000000000 +0900
+++ patches/rustc-1.44.1-src/src/llvm-project/llvm/include/llvm/Support/Compiler.h	2020-07-05 21:53:54.000000000 +0900
@@ -516,6 +516,10 @@
 // a normal global variable.
 #define LLVM_THREAD_LOCAL
 #endif
+#if defined(__APPLE__) && (defined(__arm__) || defined(__aarch64__))
+#undef LLVM_THREAD_LOCAL
+#define LLVM_THREAD_LOCAL
+#endif
 
 /// \macro LLVM_ENABLE_EXCEPTIONS
 /// Whether LLVM is built with exception support.
