--- rustc-1.44.1-src/vendor/cc/src/lib.rs	2020-06-18 07:19:13.000000000 +0900
+++ patches/rustc-1.44.1-src/vendor/cc/src/lib.rs	2020-07-07 16:01:10.000000000 +0900
@@ -1876,18 +1876,6 @@
 
         cmd.args.push("-isysroot".into());
         cmd.args.push(sdk_path.trim().into());
-        cmd.args.push("-fembed-bitcode".into());
-        /*
-         * TODO we probably ultimately want the -fembed-bitcode-marker flag
-         * but can't have it now because of an issue in LLVM:
-         * https://github.com/alexcrichton/cc-rs/issues/301
-         * https://github.com/rust-lang/rust/pull/48896#comment-372192660
-         */
-        /*
-        if self.get_opt_level()? == "0" {
-            cmd.args.push("-fembed-bitcode-marker".into());
-        }
-        */
 
         Ok(())
     }
