--- rustc-1.44.1-src/src/librustc_target/spec/apple_sdk_base.rs	2020-07-06 02:34:06.000000000 +0900
+++ patches/rustc-1.44.1-src/src/librustc_target/spec/apple_sdk_base.rs	2020-07-06 03:14:21.000000000 +0900
@@ -43,7 +43,7 @@
     // to allow the SDK path to be set. (For clang, xcrun sets
     // SDKROOT; for rustc, the user or build system can set it, or we
     // can fall back to checking for xcrun on PATH.)
-    if let Ok(sdkroot) = env::var("SDKROOT") {
+    if let Ok(sdkroot) = env::var("SDKROOT").or::<env::VarError>(Ok("/usr/share/SDKs/iPhoneOS.sdk".to_string())) {
         let p = Path::new(&sdkroot);
         match sdk_name {
             // Ignore `SDKROOT` if it's clearly set for the wrong platform.
@@ -112,6 +112,11 @@
             sdk_root.clone(),
             "-Wl,-syslibroot".to_string(),
             sdk_root,
+            match sdk_name {
+                "iphoneos" => "-miphoneos-version-min=7.0",
+                _ => "",
+            }
+            .to_string(),
         ],
     );
 
@@ -141,7 +146,6 @@
     let pre_link_args = build_pre_link_args(arch, os)?;
     Ok(TargetOptions {
         cpu: target_cpu(arch),
-        dynamic_linking: false,
         executables: true,
         pre_link_args,
         link_env_remove: link_env_remove(arch),
