--- rustc-1.44.1-src/vendor/dirs/src/lib.rs	2020-06-18 07:19:13.000000000 +0900
+++ patches/rustc-1.44.1-src/vendor/dirs/src/lib.rs	2020-07-05 14:56:23.000000000 +0900
@@ -22,7 +22,7 @@
     if #[cfg(target_os = "windows")] {
         mod win;
         use win as sys;
-    } else if #[cfg(target_os = "macos")] {
+    } else if #[cfg(any(target_os = "macos", target_os = "ios"))] {
         mod mac;
         use mac as sys;
     } else if #[cfg(target_os = "wasi")] {
