--- a/Cargo.toml
+++ b/Cargo.toml
@@ -270,7 +270,7 @@
 [dev-dependencies.test-case]
 version = "3.3.1"
 
-[target.'cfg(all(not(target_os = "windows"), not(target_os = "openbsd"), not(target_os = "aix"), not(target_os = "android"), any(target_arch = "x86_64", target_arch = "aarch64", target_arch = "powerpc64", target_arch = "riscv64")))'.dependencies.tikv-jemallocator]
+[target.'cfg(all(not(target_os = "windows"), not(target_os = "openbsd"), not(target_os = "aix"), not(target_os = "android"), not(target_os = "ios"), any(target_arch = "x86_64", target_arch = "aarch64", target_arch = "powerpc64", target_arch = "riscv64")))'.dependencies.tikv-jemallocator]
 version = "0.6.0"
 
 [target.'cfg(target_os = "windows")'.dependencies.mimalloc]
